CMP1903M Assessment 1 Base Code/Input.cs

```diff
@@ -1,40 +1,72 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace CMP1903M_Assessment_1_Base_Code
 {
+    /// <summary>
+    /// Used to accept both manual text input and text from a file
+    /// </summary>
     public class Input
     {
-        //Handles the text input for Assessment 1
-        string text = "";
-        
-        //Method: manualTextInput
-        //Arguments: none
-        //Returns: string
-        //Gets text input from the keyboard
+        /// <summary>
+        /// Used to accept text input through the command line
+        /// </summary>
+        /// <returns>string: Text input</returns>
         public string manualTextInput()
         {
-            text = "";
-            this.text = Console.ReadLine();
-            return text;
-        }
+            Console.WriteLine("Enter text, use '*' to end text entry");
+            string input = String.Empty;
+            while (true)
+            {
+                try
+                {
+                    while (!input.EndsWith('*')) // Loop until user ends entry with '*'
+                    {
+                        input += Console.ReadLine() ?? string.Empty;
+                    }
 
-        //Method: fileTextInput
-        //Arguments: string (the file path)
-        //Returns: string
-        //Gets text input from a .txt file
-        public string fileTextInput(string fileName)
+                    return input.Trim('*'); // Remove '*' and return input
+                }
+                catch (IOException e)
+                {
+                    Console.WriteLine("An IO error occured, assuming user entered nothing");
+                    return string.Empty;
+                }
+                catch (ArgumentOutOfRangeException e)
+                {
+                    Console.WriteLine("Line is too long! Try again");
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Used to get filepath from user and read contents
+        /// </summary>
+        /// <returns>string: File contents</returns>
+        public string fileTextInput()
         {
-            this.text = "";
-            using (var sr = new StreamReader(fileName))
+            while (true)
             {
-                text = sr.ReadToEnd();
+                Console.WriteLine("Enter the filepath of the text file:");
+                string fileName = Console.ReadLine() ?? string.Empty;
+
+                try
+                {
+                    // Create StreamReader to read file
+                    using (var sr = new StreamReader(fileName))
+                    {
+                        return sr.ReadToEnd().Trim('*');
+                    }

```

Comment:
Alternatively:
```suggestion
                    return File.ReadAllLines(fileName).Trim('*');
```

Reply:
Good suggestion, this seems like a better way of doing this 👍